### ISSUE TITLE
BurnInLight Setting Seems Broken

### ISSUE THREAD

**Issue by jssignore1** [DoubleDoorDevelopment/Sunburn#6]
I've been enjoying your mod, so thanks for developing it! Not sure if you're still maintaining it but figured I'd open this just in case -

I think the BurnInLight setting may be broken? In my config, I have the following settings:
```
#If light levels will be used to burn people.
burnInLight = true

#Lowest block light level that players will start to burn in at any Y level. Must enable via burnInLight. IGNORES SKY LIGHT!
#Range: 0 ~ 15
burnLightLevel = 11
```

But I can stand outside under a tree and have `Client Light: 14` for example without burning. I'm also playing with `playerMustSeeSky = true` but I tried turning that off and I still didn't seem to burn with BurnInLight. 

Based on these lines, I would expect it to work properly...
```
LayerLightEventListener blockLightingLayer = player.level.getLightEngine().getLayerListener(LightLayer.BLOCK);

if (SunBurnConfig.GENERAL.burnInLight.get() && blockLightingLayer.getLightValue(playerPos) >= SunBurnConfig.GENERAL.burnLightLevel.get())
```
so maybe something could have changed with how the player's light needs to be fetched? Just guessing since I don't have access to that code. 

Thanks in advance!

**Comment by Claycorp** [comment]
What version/game version of the mod are you using?

It's never been a very well made mod lol

**Comment by jssignore1** [comment]
Im using the latest version of the mod 3.2.3 with 1.18 game (forge)

**Comment by Claycorp** [comment]
Alright, that's modern enough to poke again. I'll take a look at it sometime soon, and possibly do a port to something newer if I never bothered if that's something you'd want too.

**Comment by jssignore1** [comment]
Oh, nice! Yeh I was thinking of making a PR with the update to 1.20 but Im new to modding so not sure how easy it would be test to make sure there are no breaking changes haha. But yeh if you're up for that I would definitely download the newest!

**Comment by jssignore1** [comment]
Hey @Claycorp , actually I just noticed another issue now that I've played a little longer. It's actually the same thing someone commented about on the curseforge page:
>I've had a lot of fun with this mod but in every world I start, the mod stops working after about 2 minecraft days and then everything goes to normal, I'm not getting damaged on sunlight no matter which settings I use.
Is this a known bug or is just something wrong with the combination of mods I'm using or something?

Since the mod stopped working completely for me after about 2 days too, I think there's some underlying issue there. Even a clean restart didnt help. So anyway, I wouldn't bother poking around with all this unless you really want to for yourself - I think I'll probably not end up using it because of that longevity problem. Sorry to file this bug without spending more time playing first! I'd kept creating new worlds for the first while with this mod so never ran into this problem til now, lol

**Comment by Claycorp** [comment]
Yeah, I can look into that too.

**Comment by jssignore1** [comment]
Yeh i confirmed its the line related to getting daytime:

```
long time = player.level().getDayTime();
...
LOGGER.info("We are at ticks:" + time + " and we are not burning.");

> We are at ticks:24267 and we are not burning.
```

I guess maybe they changed how they keep track of ticks per day, since it seems like it doesnt reset to 0 at 24000 anymore? I also found a variable `TICKS_PER_DAY = 24000;` in Minecraft's `Level` class but it doesnt seem to be used anywhere else in the source code, although maybe my search was just missing something

**Comment by jssignore1** [comment]
Ok it seems like the solution to my original problem with BurnInLight is to change line 106:
```
LayerLightEventListener blockLightingLayer = player.level.getLightEngine().getLayerListener(LightLayer.BLOCK);
```

to use `LightLayer.SKY` instead. This causes you to burn when your light level is above the specified, but only if the light is coming from outside (torch light doesnt kill you for example).

Not sure if that was the original intent of that setting/feature haha but that's what I was looking for so its working for me. 

Also for the separate ticks issue I couldnt figure it out, but tbh I wasnt going to use that setting anyway since I wanted the burn to happen at all times. Removed that one section and the mod's working great for me :D

**Comment by Claycorp** [comment]
Quick tip, if you want to reference code in a project, if you go to the page it's on and click on the line number you can link directly to it like so (click the ... menu and select copy permalink) and it should add a code snip.
https://github.com/DoubleDoorDevelopment/Sunburn/blob/e9b9a8c620a30b99b11d1caf4b0d9a3e5b6ff9a4/src/main/java/net/doubledoordev/sunburn/Sunburn.java#L130

Use shift to highlight a range of lines.

Anyway on to the actual issues here....

1. Your original issue is technically a "Working as Designed". If you enable `Block Light Burns` standing under a tree would result in no burning as skylight is not block light. `Client Light` is a maximum of the two lights. Under a tree would be 11 sky light but 0 block light and it wouldn't trigger `must see sky` regardless of the setting for example. I should probably just go over the configs again and change how this works as it's all currently confusing and just make it so you can set individual light levels for each light type.

2. Yes, the mod was broken past day 2. I must have removed the modulo that was there when I rewrote it or messed up something else. It's fixed in dev but once I find out how you reply to this for your use case of the mod I'll add the extra configs to match it and push a new build for 1.18.2 and go from there. I may also clear some of the feature ideas I had prior too like adding tags for what you can use to block the sun.

Let me know what you'd like and I'll get it fixed up and built on 1.18.2, with other versions to follow later.

**Comment by jssignore1** [comment]
Awesome, yeh Ive got a local version working for me for my preferences so feel free to fix and update it however you see fit! I think the BurnInLight setting was a little different than what I was expecting so yeh As Designed makes sense based on the expectations you described
